src/main/kotlin/com/sourcegraph/cody/edit/DocumentCodeActionHandler.kt

```diff
@@ -0,0 +1,32 @@
+package com.sourcegraph.cody.edit
+
+import com.intellij.openapi.actionSystem.DataContext
+import com.intellij.openapi.diagnostic.Logger
+import com.intellij.openapi.editor.Caret
+import com.intellij.openapi.editor.Editor
+import com.intellij.openapi.editor.actionSystem.EditorAction
+import com.intellij.openapi.editor.actionSystem.EditorActionHandler
+import com.intellij.openapi.project.DumbAware
+import com.sourcegraph.cody.autocomplete.action.CodyAction
+import com.sourcegraph.config.ConfigUtil
+import com.sourcegraph.utils.CodyEditorUtil
+
+class DocumentCodeAction : EditorAction(DocumentCodeActionHandler()), CodyAction, DumbAware
+
+class DocumentCodeActionHandler : EditorActionHandler() {
+  private val logger = Logger.getInstance(DocumentCodeActionHandler::class.java)
+
+  override fun isEnabledForCaret(editor: Editor, caret: Caret, dataContext: DataContext?): Boolean {
+    return CodyEditorUtil.isEditorValidForAutocomplete(editor) &&
+        ConfigUtil.isExperimentalInlineEditEnabled()
+  }
+
+  override fun doExecute(editor: Editor, where: Caret?, dataContext: DataContext?) {
+    val fixupService = editor.project?.getService(FixupService::class.java)
+    if (fixupService == null) {
+      logger.warn("FixupService not found")
```

Comment:
It's not really `FixupService` which was missing, but rather `Project`, right?
So maybe something like:

```kotlin
val project = editor.project
if (project == null) {
    logger.warn("No project found, cannot run DocumentCodeActionHandler::startDocumentCode")
} else {
    FixupService.getInstance(project).startDocumentCode(editor)
}
```

Reply:
Ha! Good catch; fixed.